### ISSUE TITLE
iocage jail name may not contain dots

### ISSUE THREAD

**Issue by imp1sh** [iocage/iocage#588]
Make sure to follow and check these boxes before submitting an issue! Thank you.

- [ ] Supply `iocage --version`
Version	0.9.10 2017/12/22
- [ ] Supply the commands used, along with any steps to recreate it.
```
iocage create -r 11.2-RELEASE --name a.b boot=on allow_quotas=1 quota=20G notes="Jail Template Entwicklung" vnet=on bpf=yes dhcp=on
iocage list
```
- [ ] Provide the output from the command you issued.
```
[root@jailhost001:~/hostcage # iocage create -r 11.2-RELEASE --name a.b boot=on allow_quotas=1 quota=20G notes="Jail Template Entwicklung" vnet=on bpf=yes dhcp=on
a.b successfully created!
* Starting a_b
  + Started OK
  + Configuring VNET OK
  + Starting services OK
root@jailhost001:~/hostcage # iocage list
jexec: jail "ioc-a.b" not found
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/list.py", line 114, in cli
    dataset_type, header, _long, _sort, plugin=plugins, quick=quick)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1200, in list
    exit_on_error=self.exit_on_error).list_datasets()
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 76, in list_datasets
    _all = self.list_all(ds)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_list.py", line 222, in list_all
    out = su.check_output(full_ip4_cmd)
  File "/usr/local/lib/python3.6/subprocess.py", line 336, in check_output
    **kwargs).stdout
  File "/usr/local/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['jexec', 'ioc-a.b', 'ifconfig', 'vnet0', 'inet']' returned non-zero exit status 1.]
```
- [ ] Supply what you expected the result or output to be
just list the jails
- [ ] Checked that the problem has not already been fixed on `master` if using
 a stable release.
can't find any comment containing the string "dot" in the commits after Dec 2017.

**Comment by jsegaert** [comment]
For the sake of completeness, I want to link this enhancement: **No dots allowed in jail name** #310

I imagine that weird things will happen when somebody creates two jails, one called 'foo.bar' and another 'foo_bar'.

**Comment by skarekrow** [comment]
Hmm fair point, perhaps we shouldn't allow them? Anybody else have a better suggestion?

@joachimmathes How about you?

**Comment by imp1sh** [comment]
Well dots are something very common when it comes to fully qualified domain names. I would want to use them actually for the jails.

**Comment by skarekrow** [comment]
Well IIRC jail(8) doesn't allow dots in the name. It's been a while admittedly, perhaps the existing jail detection can be extended to work with dots as well. If that's the case we should treat `_` the same as `.` imo.

**Comment by joachimmathes** [comment]
O.k, I got a little confused, since I thought this issue is gone with https://github.com/iocage/iocage/issues/310. And we still use dots in out jail names with iocage Version 0.9.10 2017/12/22. It seems to be parameter `dhcp=on` which breaks `iocage list` after jail creation.
```
# iocage create -r 11.2-RELEASE --name a.b boot=on allow_quotas=1 quota=20G notes="Jail Template Entwicklung" vnet=on bpf=yes
...
# iocage list
+-----+------+-------+--------------+-----+
| JID | NAME | STATE |   RELEASE    | IP4 |
+=====+======+=======+==============+=====+
| 3   | a.b  | up    | 11.2-RELEASE | -   |
+-----+------+-------+--------------+-----+
```
works perfectly fine.

With respect to the naming issue in general, I agree with @imp1sh. Using fully qualified domain names as jail names seems very reasonable for me.

**Comment by jsegaert** [comment]
@joachimmathes You are correct; I just wanted to be proactive and prevent that other issues caused by dots in the name linger.  I'll submit a patch to when I finish testing tomorrow-ish

**Comment by skarekrow** [comment]
Closed by #629
